src/chrome/offscreen/message-handler.ts

```diff
@@ -1,110 +1,75 @@
 import { LogsClient } from './logs-client'
-import { createMessage } from 'chrome/messages/create-message'
-import { ConnectorClient } from '@radixdlt/radix-connect-webrtc'
-import { MessagesRouter } from 'message-router'
-import { ResultAsync, errAsync, okAsync } from 'neverthrow'
-import { Queue } from 'queues/queue'
+import { errAsync, okAsync } from 'neverthrow'
 import { AppLogger, logger as appLogger } from 'utils/logger'
 import {
   Message,
   messageDiscriminator,
   MessageHandler,
   MessageHandlerOutput,
-  SendMessageWithConfirmation,
 } from '../messages/_types'
-import { LedgerResponse, isLedgerRequest } from 'ledger/schemas'
-import { sendMessage } from 'chrome/helpers/send-message'
+import { WalletConnectionClient } from './wallet-connection/wallet-connection-client'
 import { radixConnectConfig } from 'config'
-import { WalletInteractionWithOrigin } from '@radixdlt/radix-connect-schemas'
+import { Connections } from 'pairing/state/connections'
+import type { walletConnectionClientFactory } from './wallet-connection/factory'
 
 export type OffscreenMessageHandler = ReturnType<typeof OffscreenMessageHandler>
 export const OffscreenMessageHandler = (input: {
   logsClient: LogsClient
-  connectorClient: ConnectorClient
-  dAppRequestQueue: Queue<WalletInteractionWithOrigin>
-  ledgerToWalletQueue: Queue<LedgerResponse>
-  incomingWalletMessageQueue: Queue<any>
-  messageRouter: MessagesRouter
+  connectionsMap: Map<string, WalletConnectionClient>
+  walletConnectionClientFactory: walletConnectionClientFactory
   logger?: AppLogger
 }): MessageHandler => {
+  let radixConnectConfiguration: string
+  const logger = input.logger?.getSubLogger({ name: 'OffscreenMessageHandler' })
   const logsClient = input.logsClient
-  const connectorClient = input.connectorClient
-  const dAppRequestQueue = input.dAppRequestQueue
-  const ledgerToWalletQueue = input.ledgerToWalletQueue
-  const incomingWalletMessageQueue = input.incomingWalletMessageQueue
-  const messageRouter = input.messageRouter
-  const logger = input.logger || appLogger
+  const connectionsMap = input.connectionsMap
 
-  return (
-    message: Message,
-    sendMessageWithConfirmation: SendMessageWithConfirmation,
-    tabId?: number,
-  ): MessageHandlerOutput => {
+  return (message: Message): MessageHandlerOutput => {
     switch (message?.discriminator) {
-      case messageDiscriminator.walletMessage: {
-        if (isLedgerRequest(message.data)) {
-          logger.debug('🪪 -> 📒: walletToLedgerSubject', message.data)
+      case messageDiscriminator.setConnections: {
+        logger?.debug('setConnections', { message })
```

Comment:
Users are posting connector extension logs in public channels. We should not log the connection password

Reply:
something in between is smart enough to do this 
![Screenshot 2024-01-03 at 15 26 15](https://github.com/radixdlt/connector-extension/assets/119939994/880812b9-03d8-4ce4-9962-5dabca009014)